corehq/apps/app_manager/views.py

```diff
@@ -908,9 +909,19 @@ def get_details(case_type_):
         def _report_to_config(report):
             return {
                 'report_id': report._id,
-                'title': report.title
+                'title': report.title,
+                'charts': [chart for chart in report.configured_charts if chart['type'] == 'multibar'],
             }
         all_reports = ReportConfiguration.by_domain(app.domain)
+        for report in all_reports:
+            for i in range(len(report.configured_charts)):
+                chart_id = report.configured_charts[i].get('chart_id')
+                if not chart_id:
+                    report.configured_charts[i]['chart_id'] = (
+                        report.configured_charts[i].get('title') or ''
+                    ) + '-' + uuid.uuid4().hex
+            report.save()
```

Comment:
I think we definitely want to save it. I was just saying that as long as you generate it deterministically based on other information in the `ChartSpec` then you don't have to actually save it back until something else changes (which would require a save anyways). Using `.wrap` and the wrapped objects will ensure it's always available whenever you reference it